### PR TITLE
Add docker context to codebuild module and change dockerhub env var f…

### PR DESCRIPTION
**Pull request by gds-ahine**
…or credentials to use SECRETS_MANAGER type

### CODE REVIEW

File: codebuild/codebuild_project_container_build_docker_hub/code_build_terraform.tf [hunk 1]
```diff
@@ -38,17 +38,24 @@ resource "aws_codebuild_project" "code_pipeline_container_build_docker_hub" {
     environment_variable {
       name  = "DOCKERHUB_USERNAME"
       value = var.docker_hub_username
+      type  = SECRETS_MANAGER
```
Comment (danjoneslf) [inline codebuild/codebuild_project_container_build_docker_hub/code_build_terraform.tf:41]: What are you going to pass into these vars? The value in secrets manager is JSON with a username and password property. I think if you pass in a secrets manager secret you have to then pass the path to the value you want to pull out.

Comment (gds-ahine) [reply codebuild/codebuild_project_container_build_docker_hub/code_build_terraform.tf]: I see, this is already being parsed out from the calling module. I've added another commit to remove this for now. Would adding a `registry_credential` resolve this problem ?

Comment (gds-ahine) [reply codebuild/codebuild_project_container_build_docker_hub/code_build_terraform.tf]: Still not having much joy with this (see latest commit) maybe it's the jsondecode function, the secrets are still output in plaintext when running a `terraform plan`.

**Review by detnon — DISMISSED** [review]
Looks good

File: codebuild/codebuild_ECR_push/code_build_ecr_container.yml [hunk 2]
```diff
@@ -3,17 +3,17 @@ version: 0.2
 phases:
   pre_build:
     commands:
+      - echo Assuming role into target AWS account for deployment
+      - role_arn="arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ROLE_NAME}"
+      - CODEPIPELINE_DEPLOYER_ROLE=`aws sts assume-role --role-arn $role_arn --role-session-name codepipeline-ecr-task`
+      - export AWS_ACCESS_KEY_ID=`echo "${CODEPIPELINE_DEPLOYER_ROLE}" | jq -r '.Credentials.AccessKeyId'`
+      - export AWS_SECRET_ACCESS_KEY=`echo "${CODEPIPELINE_DEPLOYER_ROLE}" | jq -r '.Credentials.SecretAccessKey'`
+      - export AWS_SESSION_TOKEN=`echo "${CODEPIPELINE_DEPLOYER_ROLE}" | jq -r '.Credentials.SessionToken'`
```
Comment (akinnane) [inline codebuild/codebuild_ECR_push/code_build_ecr_container.yml:11]: Could all of this assume role code be moved into a script that can be stored on the base image and shared between modules?

Comment (gds-ahine) [reply codebuild/codebuild_ECR_push/code_build_ecr_container.yml:11]: Good shout, can do 👍

Comment (danjoneslf) [reply codebuild/codebuild_ECR_push/code_build_ecr_container.yml:11]: It's already there. It's in `bin/assume-role.sh`

File: codebuild/codebuild_get_changed_files/codebuild_get_changed_files.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+phases:
+  build:
+    commands:
+      - echo '[INFO] Reading artifacts...'
+      - pwd
+      - ls
+      - cd $TERRAFORM_DIR
+      - ls
+      - ./github_rest_get_pr_changed_files_list.sh $REPO_NAME /opt/changed_files.json
```
Comment (danjoneslf) [inline codebuild/codebuild_get_changed_files/codebuild_get_changed_files.yml:14]: I don't think we want to do that here. There's already a module for it here: https://github.com/alphagov/cyber-security-shared-terraform-modules/blob/master/codebuild/codebuild_git_diff/code_build_git_diff.yml

We just want to do this once and pass this around as an artifact.

**Pull request closed without merging**